CMakeLists.txt

```diff
@@ -306,6 +314,15 @@ else()
 	endif (MSVC)
 endif()
 
+if (NOT MSVC)
+	if (abi STREQUAL "c++03")
+		add_definitions(-std=${abi})
+	elseif (abi STREQUAL "c++11" OR abi STREQUAL "c++14")
+		# Also silence auto_ptr because it throws lots of warnings.
+		add_definitions(-std=${abi} -Wno-deprecated-declarations)
```

Comment:
`auto_ptr` is deprecated in C++11, and libtorrent::session uses it for dispatch handlers. That generates a ton of deprecation warnings.